### ISSUE TITLE
Calendar: The app crashes on iOS 10

### ISSUE THREAD

**Issue by arturs105** [toggl/mobileapp#3038]
### Steps to reproduce:
1. Run the app on an iOS 10 simulator
2. Log in with test.account.2
3. 💥

**Comment by juanlaube** [comment]
Crash happens in CalendarService.cs, line 33: `var calendarItems = eventStore.EventsMatching(predicate)`

**Comment by juanlaube** [comment]
From application output:

```
2018-08-22 11:36:48.151 Toggl.Daneel[68827:14433083] error: * Assertion at ../../../../../mono/mini/debugger-agent.c:4856, condition `array->len == 1' not met
2018-08-22 11:36:48.151 Toggl.Daneel[68827:14433083] critical: Stacktrace:

2018-08-22 11:36:48.151 Toggl.Daneel[68827:14433083] critical:   at <unknown> <0xffffffff>
2018-08-22 11:36:48.152 Toggl.Daneel[68827:14433083] critical:   at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain (int,string[],intptr,intptr) [0x0005c] in <e4467505ca564435940d4efa0333f9fa>:0
2018-08-22 11:36:48.152 Toggl.Daneel[68827:14433083] critical:   at UIKit.UIApplication.Main (string[],intptr,intptr) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/11.14.0.13/src/Xamarin.iOS/UIKit/UIApplication.cs:79
2018-08-22 11:36:48.152 Toggl.Daneel[68827:14433083] critical:   at UIKit.UIApplication.Main (string[],string,string) [0x0002c] in /Library/Frameworks/Xamarin.iOS.framework/Versions/11.14.0.13/src/Xamarin.iOS/UIKit/UIApplication.cs:63
2018-08-22 11:36:48.152 Toggl.Daneel[68827:14433083] critical:   at Toggl.Daneel.Application.Main (string[]) [0x00000] in /Users/juanlaube/Developer/Toggl/mobileapp/Toggl.Daneel/Startup/Main.cs:8
2018-08-22 11:36:48.153 Toggl.Daneel[68827:14433083] critical:   at (wrapper runtime-invoke) <Module>.runtime_invoke_void_object (object,intptr,intptr,intptr) [0x00051] in <f56cbb852b9043d383d871ff5945c6d8>:0
2018-08-22 11:36:48.153 Toggl.Daneel[68827:14433083] critical: 
Native stacktrace:

2018-08-22 11:36:48.174 Toggl.Daneel[68827:14433083] critical: 	0   Toggl.Daneel                        0x00000001003fb0a4 mono_handle_native_crash + 244
2018-08-22 11:36:48.175 Toggl.Daneel[68827:14433083] critical: 	1   libsystem_platform.dylib            0x000000010686ff5a _sigtramp + 26
2018-08-22 11:36:48.175 Toggl.Daneel[68827:14433083] critical: 	2   ???                                 0x000000000000ffff 0x0 + 65535
2018-08-22 11:36:48.175 Toggl.Daneel[68827:14433083] critical: 	3   libsystem_c.dylib                   0x00000001065d988f abort + 127
2018-08-22 11:36:48.175 Toggl.Daneel[68827:14433083] critical: 	4   Toggl.Daneel                        0x0000000100614d50 _ZL12log_callbackPKcS0_S0_iPv + 64
2018-08-22 11:36:48.175 Toggl.Daneel[68827:14433083] critical: 	5   Toggl.Daneel                        0x00000001005e3cc3 monoeg_g_logv + 83
2018-08-22 11:36:48.176 Toggl.Daneel[68827:14433083] critical: 	6   Toggl.Daneel                        0x00000001005e3fb7 monoeg_assertion_message + 199
2018-08-22 11:36:48.176 Toggl.Daneel[68827:14433083] critical: 	7   Toggl.Daneel                        0x0000000100358e24 get_this_async_id + 420
2018-08-22 11:36:48.176 Toggl.Daneel[68827:14433083] critical: 	8   Toggl.Daneel                        0x000000010034b766 process_breakpoint + 726
2018-08-22 11:36:48.176 Toggl.Daneel[68827:14433083] critical: 	9   Toggl.Daneel                        0x000000010034b41e debugger_agent_breakpoint_from_context + 158
2018-08-22 11:36:48.176 Toggl.Daneel[68827:14433083] critical: 	10  ???                                 0x000000011878307f 0x0 + 4705497215
2018-08-22 11:36:48.176 Toggl.Daneel[68827:14433083] critical: 	11  ???                                 0x000000011a92fe55 0x0 + 4740808277
2018-08-22 11:36:48.177 Toggl.Daneel[68827:14433083] critical: 	12  ???                                 0x000000011a92c22f 0x0 + 4740792879
2018-08-22 11:36:48.177 Toggl.Daneel[68827:14433083] critical: 	13  ???                                 0x000000011a92b4f3 0x0 + 4740789491
2018-08-22 11:36:48.177 Toggl.Daneel[68827:14433083] critical: 	14  ???                                 0x000000011a0530d3 0x0 + 4731515091
2018-08-22 11:36:48.177 Toggl.Daneel[68827:14433083] critical: 	15  Toggl.Daneel                        0x000000010040ddf3 mono_jit_runtime_invoke + 1443
2018-08-22 11:36:48.177 Toggl.Daneel[68827:14433083] critical: 	16  Toggl.Daneel                        0x00000001004fdabf mono_runtime_invoke_checked + 127
2018-08-22 11:36:48.178 Toggl.Daneel[68827:14433083] critical: 	17  Toggl.Daneel                        0x0000000100501533 mono_runtime_invoke + 83
2018-08-22 11:36:48.178 Toggl.Daneel[68827:14433083] critical: 	18  Toggl.Daneel                        0x00000001001945d5 _ZL31native_to_managed_trampoline_87P11objc_objectP13objc_selectorPP11_MonoMethodP13UIApplicationP12NSDictionaryj + 949
2018-08-22 11:36:48.178 Toggl.Daneel[68827:14433083] critical: 	19  Toggl.Daneel                        0x0000000100194808 -[AppDelegate application:didFinishLaunchingWithOptions:] + 72
2018-08-22 11:36:48.178 Toggl.Daneel[68827:14433083] critical: 	20  UIKit                               0x0000000102bce9b7 -[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:] + 299
2018-08-22 11:36:48.178 Toggl.Daneel[68827:14433083] critical: 	21  UIKit                               0x0000000102bd029c -[UIApplication _callInitializationDelegatesForMainScene:transitionContext:] + 4119
2018-08-22 11:36:48.178 Toggl.Daneel[68827:14433083] critical: 	22  UIKit                               0x0000000102bd65e4 -[UIApplication _runWithMainScene:transitionContext:completion:] + 1709
2018-08-22 11:36:48.179 Toggl.Daneel[68827:14433083] critical: 	23  UIKit                               0x0000000102bd37f3 -[UIApplication workspaceDidEndTransaction:] + 182
2018-08-22 11:36:48.179 Toggl.Daneel[68827:14433083] critical: 	24  FrontBoardServices                  0x00000001095365f6 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 24
2018-08-22 11:36:48.179 Toggl.Daneel[68827:14433083] critical: 	25  FrontBoardServices                  0x000000010953646d -[FBSSerialQueue _performNext] + 186
2018-08-22 11:36:48.179 Toggl.Daneel[68827:14433083] critical: 	26  FrontBoardServices                  0x00000001095367f6 -[FBSSerialQueue _performNextFromRunLoopSource] + 45
2018-08-22 11:36:48.179 Toggl.Daneel[68827:14433083] critical: 	27  CoreFoundation                      0x0000000104efbc01 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
2018-08-22 11:36:48.179 Toggl.Daneel[68827:14433083] critical: 	28  CoreFoundation                      0x0000000104ee10cf __CFRunLoopDoSources0 + 527
2018-08-22 11:36:48.180 Toggl.Daneel[68827:14433083] critical: 	29  CoreFoundation                      0x0000000104ee05ff __CFRunLoopRun + 911
2018-08-22 11:36:48.180 Toggl.Daneel[68827:14433083] critical: 	30  CoreFoundation                      0x0000000104ee0016 CFRunLoopRunSpecific + 406
2018-08-22 11:36:48.180 Toggl.Daneel[68827:14433083] critical: 	31  UIKit                               0x0000000102bd208f -[UIApplication _run] + 468
2018-08-22 11:36:48.180 Toggl.Daneel[68827:14433083] critical: 	32  UIKit                               0x0000000102bd8134 UIApplicationMain + 159
2018-08-22 11:36:48.180 Toggl.Daneel[68827:14433083] critical: 	33  ???                                 0x0000000119ad171e 0x0 + 4725741342
2018-08-22 11:36:48.180 Toggl.Daneel[68827:14433083] critical: 	34  ???                                 0x0000000119ad14b3 0x0 + 4725740723
2018-08-22 11:36:48.181 Toggl.Daneel[68827:14433083] critical: 
=================================================================
Got a SIGABRT while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================
```

**Comment by juanlaube** [comment]
Closed by #3040
